irods_consortium_continuous_integration_build_hook.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import print_function
+
+import multiprocessing
+import optparse
+import os
+import tempfile
+
+import irods_python_ci_utilities
+
+
+def get_build_prerequisites_all():
+    pass;
+
+def get_build_prerequisites_apt():
+    pass;
+
+def get_build_prerequisites_yum():
+    pass;
+
+def get_build_prerequisites():
+    dispatch_map = {
+        'Ubuntu': get_build_prerequisites_apt,
+        'Centos': get_build_prerequisites_yum,
+        'Centos linux': get_build_prerequisites_yum,
+        'Opensuse ': get_build_prerequisites_yum,
+    }
+    try:
+        return dispatch_map[irods_python_ci_utilities.get_distribution()]()
+    except KeyError:
+        irods_python_ci_utilities.raise_not_implemented_for_distribution()
+
+def install_build_prerequisites_apt():
+    if irods_python_ci_utilities.get_distribution() == 'Ubuntu': # cmake from externals requires newer libstdc++ on ub12
+        if irods_python_ci_utilities.get_distribution_version_major() == '12':
+            irods_python_ci_utilities.install_os_packages(['python-software-properties'])
+            irods_python_ci_utilities.subprocess_get_output(['sudo', 'add-apt-repository', '-y', 'ppa:ubuntu-toolchain-r/test'], check_rc=True)
+            irods_python_ci_utilities.install_os_packages(['libstdc++6'])
+    #irods_python_ci_utilities.install_os_packages(get_build_prerequisites())
+
+def install_build_prerequisites_yum():
+    if irods_python_ci_utilities.get_distribution() == 'Centos' or irods_python_ci_utilities.get_distribution() == 'Centos linux':
+        irods_python_ci_utilities.subprocess_get_output(['sudo', 'systemctl', 'restart', 'autofs'], check_rc=True)
+        if irods_python_ci_utilities.get_distribution_version_major() == '6':
+            os.environ['CC'] = '/opt/rh/devtoolset-6/root/usr/bin/gcc'
+            os.environ['CXX'] = '/opt/rh/devtoolset-6/root/usr/bin/g++'
+            os.environ['PATH'] = '/opt/rh/devtoolset-6/root/usr/bin' + os.pathsep + os.environ['PATH']
+    #irods_python_ci_utilities.install_os_packages(get_build_prerequisites())
```

Comment:
Given that we will not be supporting Cen6 or Ub12, those sections can be removed. This effectively eliminates install_build_prerequisites_* except for lines 41 and 42 which is where the action happens. Can we move these two lines to main() or to a well-named function and delete the rest?